From 6e2fd1095cead1e7e52e953f29d03f537e20475d Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Wed, 11 Sep 2024 19:31:04 +0800
Subject: [PATCH 4/4] Revert "dril: use the super fallback path for software
 fallback"

This reverts commit 06d417af80bc1f171cadc338e63a7aa75c877754.

Ref: https://gitlab.freedesktop.org/mesa/mesa/-/issues/11839
Ref: https://gitlab.freedesktop.org/mesa/mesa/-/issues/11840
---
 src/gallium/targets/dril/dril_target.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/src/gallium/targets/dril/dril_target.c b/src/gallium/targets/dril/dril_target.c
index 95007e22674..043700e5cca 100644
--- a/src/gallium/targets/dril/dril_target.c
+++ b/src/gallium/targets/dril/dril_target.c
@@ -453,13 +453,8 @@ drilCreateNewScreen(int scrn, int fd,
                     const __DRIconfig ***driver_configs, void *data)
 {
    const __DRIconfig **configs = init_dri2_configs(fd);
-   if (!configs && fd == -1) {
-      // otherwise set configs to point to our config list
-      configs = calloc(ARRAY_SIZE(drilConfigs) + 1, sizeof(void *));
-      for (int i = 0; i < ARRAY_SIZE(drilConfigs); i++) {
-         configs[i] = mem_dup(&drilConfigs[i], sizeof(drilConfigs[i]));
-      }
-   }
+   if (!configs)
+      return NULL;
 
    // outpointer it
    *driver_configs = configs;
-- 
2.47.0

